core/cap-0056.md

```diff
@@ -51,12 +51,14 @@ the hash stored in `TransactionResultPairV2`.
 
 #### Host functions
 ```rust
-/// Will save val to a list that will be returned to core so the val can be
-/// put into TransactionMetaV3.events with type ContractEventType::CONTRACT.
-func $contract_event(param $val i64) (result i64)
+/// Will save val and topics to a list that will be returned to core so ContractEvent can be
+/// populated and put into TransactionMetaV3.events with type ContractEventType::CONTRACT.
+/// topics is expected to be a SCVec with length <= 4 that cannot contain Vecs, Maps,
+/// or Binaries > 32 bytes
+func $contract_event(param $topics i64)(param $val i64) (result i64)
 
 /// Internal host function. Same as $system_event but will use ContractEventType::SYSTEM instead
-func $system_event(param $val i64) (result i64)
+func $system_event(param $topics i64)(param $val i64) (result i64)
```

Comment:
The term topics is better 👍. Could we rename val to event, or message, to give it more color similar to how we're naming topic?

Reply:
Or make it consistent with `data` that is further down.

Reply:
I renamed `val` to `event`

Reply:
Should the `data` be renamed to `event` further down too?

Reply:
Sorry, my comment was wrong. I actually renamed `val` to `data`.

Reply:
Love it. 👍🏻